src/test/e2e/api/client/feature.optimal304.e2e.test.ts

```diff
@@ -1,220 +1,314 @@
         const res = await app.request
             .get('/api/client/features')
+            .set('Authorization', devTokenSecret)
             .expect('Content-Type', /json/)
             .expect(200);
 
-        if (etagVariant.feature_enabled) {
-            expect(res.headers.etag).toBe(`"76d8bb0e:16:${etagVariant.name}"`);
-            expect(res.body.meta.etag).toBe(
-                `"76d8bb0e:16:${etagVariant.name}"`,
-            );
+        if (enabled) {
+            expect(res.headers.etag).toBe(`"76d8bb0e:13:${name}"`);
+            expect(res.body.meta.etag).toBe(`"76d8bb0e:13:${name}"`);
         } else {
-            expect(res.headers.etag).toBe('"76d8bb0e:16"');
-            expect(res.body.meta.etag).toBe('"76d8bb0e:16"');
+            expect(res.headers.etag).toBe('"76d8bb0e:13"');
+            expect(res.body.meta.etag).toBe('"76d8bb0e:13"');
         }
     });
 
-    test(`returns ${etagVariant.feature_enabled ? 200 : 304} for pre-calculated hash${etagVariant.feature_enabled ? ' because hash changed' : ''}`, async () => {
+    test(`returns ${enabled ? 200 : 304} for pre-calculated hash${enabled ? ' because hash changed' : ''} (dev env token)`, async () => {
         const res = await app.request
             .get('/api/client/features')
-            .set('if-none-match', '"76d8bb0e:16"')
-            .expect(etagVariant.feature_enabled ? 200 : 304);
-
-        if (etagVariant.feature_enabled) {
-            expect(res.headers.etag).toBe(`"76d8bb0e:16:${etagVariant.name}"`);
-            expect(res.body.meta.etag).toBe(
-                `"76d8bb0e:16:${etagVariant.name}"`,
-            );
+            .set('Authorization', devTokenSecret)
+            .set('if-none-match', '"76d8bb0e:13"')
+            .expect(enabled ? 200 : 304);
+
+        if (enabled) {
+            expect(res.headers.etag).toBe(`"76d8bb0e:13:${name}"`);
+            expect(res.body.meta.etag).toBe(`"76d8bb0e:13:${name}"`);
         }
     });
 
-    test('returns 200 when content updates and hash does not match anymore', async () => {
-        await app.services.featureToggleService.createFeatureToggle(
-            'default',
-            {
-                name: `featureNew304${apendix}`,
-                description: 'the #1 feature',
-            },
-            TEST_AUDIT_USER,
-        );
+    test('creating a new feature does not modify etag', async () => {
+        await app.createFeature('new');
         await app.services.configurationRevisionService.updateMaxRevisionId();
 
-        const res = await app.request
+        await app.request
             .get('/api/client/features')
-            .set('if-none-match', 'ae443048:16')
+            .set('Authorization', devTokenSecret)
+            .set('if-none-match', `"76d8bb0e:13${enabled ? `:${name}` : ''}"`)
+            .expect(304);
+    });
+
+    test('a token with all envs should get the max id regardless of the environment', async () => {
```

Comment:
Test for all environments token (`'*'`). I didn't know this was supported. At least invalid from the UI, but it's still part of our code, but it behaves as a fallback to the default environment: https://github.com/Unleash/unleash/blob/18bea2538318cb7aaa26e6583a65547a18eeae16/src/lib/features/frontend-api/global-frontend-api-cache.ts#L142-L146

I think this approach of defaulting to the max revision id is also valid and covers either case.